addons/web_editor/static/lib/odoo-editor/src/utils/utils.js

```diff
@@ -1217,14 +1217,12 @@ export function isFontSize(node, props) {
  * @returns {boolean}
  */
 export function isSelectionFormat(editable, format) {
-    const selectedText = getSelectedNodes(editable)
+    const selectedNodes = getSelectedNodes(editable)
         .filter(n => n.nodeType === Node.TEXT_NODE && n.nodeValue.trim().length);
     const isFormatted = formatsSpecs[format].isFormatted;
-    if (selectedText.length) {
-        return selectedText.every(n => isFormatted(n.parentElement, editable))
-    } else {
-        return isFormatted(closestElement(editable.ownerDocument.getSelection().anchorNode), editable);
-    }
+    selectedNodes.push(closestElement(editable.ownerDocument.getSelection().anchorNode));
+    selectedNodes.push(closestElement(editable.ownerDocument.getSelection().focusNode));
+    return selectedNodes.every(n => isFormatted(n.parentElement, editable))
```

Comment:
I suggest to add tests for these usecases. They can be added to the existing file with formats tests.

Reply:
done